enhancements/platform-validation.md

```diff
@@ -0,0 +1,195 @@
+
+Deploy using Kepler Operator. Follow guide [here](https://github.com/sustainable-computing-io/kepler-operator/blob/v1alpha1/README.md).
+
+Deploy from Kepler Source Code. Follow guide [here](https://sustainable-computing.io/installation/kepler/).
+
+All the above deployment scenarios related validation should be automated. i.e. locate at $Kepler_ROOT/hack/platform-validation-deploy-from-xxx.sh.
+
+Step 3: Run test cases and get test result.
+
+Step 4: Generate test report and save to somewhere.
+
+Step 5: Delete the local cluster and recover the orignal environment on the test node.
+
+### Implementation Details/Notes/Constraints
+
+Our validation test is local-dev based, in another word, [Kind](https://kind.sigs.k8s.io/) based. 
+
+Since the Kind is Kubernetes cluster running in Docker container, it may be difficult for us to directly check the Kepler exposed metrics on port 9102.
+
+The easier way is to deploy Prometheus service and Kepler related Service Monitor in the cluster. Expose Prometheus service as `NodePort` type.
+Codes/scripts could then retrieve what we needed data from Prometheus using either REST API calling or PromQL queries.
+
+A typical but not only way is could be found in `Test Plan` section below.
+
+### Risks and Mitigations
+
+
+### Drawbacks
+
+
+## Design Details
+
+### Test Plan
+
+The initial test case is only for CPU architecture recognition.
+
+When we check the `cpu_architecture` label exported by Kepler, we do below things:
+
+1. Deploy Kepler with Prometheus.
+
+```
+make build-manifest OPTS="PROMETHEUS_DEPLOY"
+```
+
+2. Change the Prometheus service as `NodePort` type and get the accessible port.
+
+```
+kubectl patch svc prometheus-k8s -n monitoring --type='json' -p '[{"op":"replace","path":"/spec/type","value":"NodePort"}]'
+```
+```
+$kubectl get svc prometheus-k8s -n monitoring
+NAME             TYPE       CLUSTER-IP    EXTERNAL-IP   PORT(S)                         AGE
+prometheus-k8s   NodePort   10.96.205.9   <none>        9090:32517/TCP,8080:30331/TCP   5d9h
+```
+
+Tools such as `awk` could retrieve the accessible port:
+```
+$kubectl get svc prometheus-k8s -n monitoring | grep 9090 | awk '{print $5}' | awk -F ',' '{print $1}' | awk -F '/' '{print substr($1,6)}'
+32517
+```
+
+3. Get the test node's IP.
+
+```
+$ kubectl get nodes -o wide
+NAME                 STATUS   ROLES           AGE     VERSION   INTERNAL-IP   EXTERNAL-IP   OS-IMAGE             KERNEL-VERSION      CONTAINER-RUNTIME
+kind-control-plane   Ready    control-plane   5d10h   v1.25.3   172.19.0.2    <none>        Ubuntu 22.04.1 LTS   5.19.0-41-generic   containerd://1.6.9
+```
+
+```
+$ kubectl get nodes -o wide |grep control-plane | awk '{print $6}'
+172.19.0.2
+```
+
+4. Retrieve the metrics with `cpu_architecture` label in Prometheus.
+```
+$ curl --noproxy "*" -s http://172.19.0.2:32517/api/v1/label/cpu_architecture/values | jq -r ".data[]" | sort
+Sapphire Rapids
+```
+
+5. Check the actual `cpu_architure` info on the node:
```

Comment:
is there any case that `cpu_architure` mis match? or say if we use kind. can we use uname to get CPU arch and do the compare works? as we can even deploy a kube cluster on the server... as we have ssh access right? or the steps is without SSH connection?

Reply:
> is there any case that `cpu_architure` mis match? or say if we use kind. can we use uname to get CPU arch and do the compare works? as we can even deploy a kube cluster on the server... as we have ssh access right? or the steps is without SSH connection?

hmm, this test case is something like white box test, because the `cpu_architecture` label is added by our Kepler metrics collector code. On Prometheus point of view(sometimes also customer point of view), such label stands for the hardware platform, so the check logic should be consistent with our code, otherwise, as Marcelo said, there is a bug...
Think about this: when customers check metrics in Prom, they see various of "unknown" `cpu_architecture` labels...